### PR TITLE
Updates Design Docs for v0.2

### PR DESCRIPTION
**Pull request by danehans**
Updates design docs based on the current implementation (as of the v0.2).

Signed-off-by: danehans <daneyonhansen@gmail.com>

### CODE REVIEW

File: docs/design/gatewayapi-translator.md [hunk 1]
```diff
@@ -1,35 +1,38 @@
 # Gateway API Translator Design
 
+The Gateway API translates external resources, e.g. GatewayClass, from the configured Provider's Resource Watcher to the
```
Comment (arkodg) [inline docs/design/gatewayapi-translator.md:3]: should we just stick to `Provider` and rm `Resource Watcher` ?

File: docs/design/gatewayapi-translator.md [hunk 2]
```diff
@@ -172,28 +139,32 @@ spec:
...
-- each ParentRef for each Route (route.status.parents)
+- The managed GatewayClass (`gatewayclass.status.conditions`).
+- Each managed Gateway, based on its Listeners' status (`gateway.status.conditions`). For the Kubernetes provider, the
+  Envoy Deployment and Service status are also included to calculate Gateway status.
+- Listeners for each Gateway (`gateway.status.listeners`).
+- The ParentRef for each Route (`route.status.parents`).
 
-The Gateway API translator will take the approach of populating status on the resources themselves as they're being processed, and then passing those statuses off to another component to persist the updates to the Kubernetes API or other backend.
+The Gateway API translator is responsible for calculating status conditions while translating Gateway API resources to
+the IR and publishing status over the [message bus][]. The Status Manager subscribes to these status messages and
```
Comment (arkodg) [inline docs/design/gatewayapi-translator.md:154]: message bus needs a link

File: docs/design/gatewayapi-translator.md (hunk 1 above)
Comment (danehans) [reply docs/design/gatewayapi-translator.md]: Ack.

File: docs/design/gatewayapi-translator.md (hunk 2 above)
Comment (danehans) [reply docs/design/gatewayapi-translator.md:154]: @arkodg the link is in [L251](https://github.com/envoyproxy/gateway/pull/645/files#diff-9525aca7f46ffbe1952dae4f2c5be610a6048e8c65af05dd651858ac3ce1e16fR251). I generated the docs, ran them locally, and verified the link to `watching.md` works.

**Review by arkodg — APPROVED** [review]
LGTM thanks, hoping @skriss can also review since some of the changes are around the Gateway API Translator design doc

**Review by Xunzhuo — APPROVED** [review]
Looks good:)

**Review by skriss — APPROVED** [review]
LGTM, good updates 👍

**Pull request merged**